### DISCUSSION TITLE
Where is the documentation for Realtime Database inside Cloud Function?

### DISCUSSION THREAD

**Discussion by sorcamarian** [firebase/functions-samples#1121] [Q&A] [answered]
Where is the documentation for Realtime Database inside Cloud Function?

**Comment by puf** [comment]
https://firebase.google.com/docs/functions/database-events

**Comment by sorcamarian** [comment]
I was searching for basic things like how to get all the values of a path

`db.ref("dataTables").once("value", function(snapshot) {
   console.log(snapshot.val() )
})`

**Reply by puf** [reply, marked as answer]
The only thing that's specific to Cloud Functions there is how you get a reference to the database, for the rest it's just the regular JavaScript/Node.js SDK.

For getting a DB reference, you can either get it from the event that triggered the function (through `event.data.ref.root`) or by initializing the Admin SDK and then getting it from there.

If you tried this and can't get it to work, I recommend posting a [minimal repro](http://stackoverflow.com/help/mcve) to one of the more support oriented communities, such as Stack Overflow or /r/Firebase on Reddit.
